### PR TITLE
[REVIEW] removing old db code  [skip-ci]

### LINKED ISSUE #1193: [BUG] Drop DB from cugraph (triggering debug build linker issues)
We'll keep iterating on that in a sperate repo for now. No need to keep maintaining and building it in cuGraph at the moment.
- `cpp/src/db` folder 
- `libcypher-parser` dependency

### PR DESCRIPTION
**Pull request by BradReesWork**
Removing old db code that is no longer needed

#close #1193

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -430,7 +413,7 @@ target_include_directories(cugraph
 # - link libraries --------------------------------------------------------------------------------
 
 target_link_libraries(cugraph PRIVATE
-    gunrock cublas cusparse curand cusolver cudart cuda ${LIBCYPHERPARSER_LIBRARY} ${MPI_CXX_LIBRARIES} ${NCCL_LIBRARIES})
+    gunrock cublas cusparse curand cusolver cudart cuda ${MPI_CXX_LIBRARIES} ${NCCL_LIBRARIES})
```
Comment (afender) [inline cpp/CMakeLists.txt:416]: unrelated but ${MPI_CXX_LIBRARIES} caught my attention, not sure it is really needed here

**Review by afender — APPROVED** [review]

File: cpp/CMakeLists.txt (hunk 1 above)
Comment (BradReesWork) [reply cpp/CMakeLists.txt]: dropped MPI_CXX_LIBRARIES

**Review by jwyles — APPROVED** [review]
Looks good to me.

**Comment by BradReesWork** [comment]
rerun tests

**Pull request merged**